packages/driver/cypress/integration/commands/sessions/manager_spec.ts

```diff
@@ -0,0 +1,354 @@
+        'session_1': {
+          id: 'session_1',
+          setup: () => {},
+          hydrated: false,
+        },
+        'session_2': {
+          id: 'session_2',
+          setup: () => {},
+          hydrated: true,
+        },
+      }
+
+      const cySpy = cy.stub(cy, 'state').callThrough().withArgs('activeSessions').returns(existingSessions)
+
+      const sessionsManager = new SessionsManager(CypressInstance, cy)
+
+      sessionsManager.clearActiveSessions()
+      const calls = cySpy.getCalls()
+
+      expect(cySpy).to.be.calledTwice
+      expect(calls[1].args[1]).to.be.instanceOf(Object)
+      expect(calls[1].args[1]).to.haveOwnProperty('session_1')
+      expect(calls[1].args[1].session_1).to.haveOwnProperty('hydrated', false)
+      expect(calls[1].args[1]).to.haveOwnProperty('session_2')
+      expect(calls[1].args[1].session_2).to.haveOwnProperty('hydrated', false)
+    })
+  })
+
+  describe('.mapOrigins()', () => {
+    it('maps when requesting all origins', async () => {
+      const sessionsManager = new SessionsManager(CypressInstance, cy)
+
+      const allOrigins = ['https://example.com', baseUrl, 'http://foobar.com', 'http://foobar.com']
+      const sessionsSpy = cy.stub(sessionsManager, 'getAllHtmlOrigins').resolves(allOrigins)
+
+      const origins = await sessionsManager.mapOrigins('*')
+
+      expect(origins).to.deep.eq(['https://example.com', baseUrl, 'http://foobar.com'])
+      expect(sessionsSpy).to.be.calledOnce
+    })
+
+    it('maps when requesting the current origin', async () => {
+      const sessionsManager = new SessionsManager(CypressInstance, cy)
+      const sessionsSpy = cy.stub(sessionsManager, 'getAllHtmlOrigins')
+      const origins = await sessionsManager.mapOrigins('currentOrigin')
+
+      expect(origins).to.deep.eq([baseUrl])
+      expect(sessionsSpy).not.to.be.called
+    })
+
+    it('maps when requesting a specific origin', async () => {
+      const sessionsManager = new SessionsManager(CypressInstance, cy)
+      const sessionsSpy = cy.stub(sessionsManager, 'getAllHtmlOrigins')
+      const origins = await sessionsManager.mapOrigins('https://example.com/random_page?1')
+
+      expect(origins).to.deep.eq(['https://example.com'])
+      expect(sessionsSpy).not.to.be.called
+    })
+
+    it('maps when requesting a list of origins', async () => {
+      const sessionsManager = new SessionsManager(CypressInstance, cy)
+
+      const allOrigins = ['https://example.com', baseUrl, 'http://foobar.com', 'http://foobar.com']
+      const sessionsSpy = cy.stub(sessionsManager, 'getAllHtmlOrigins').resolves(allOrigins)
+
+      const origins = await sessionsManager.mapOrigins(['*', 'https://other.com'])
+
+      expect(origins).to.deep.eq(['https://example.com', baseUrl, 'http://foobar.com', 'https://other.com'])
+      expect(sessionsSpy).to.be.calledOnce
+    })
+  })
+
+  describe('._setStorageOnOrigins()', () => {})
```

Comment:
I assume you meant for this describe to wrap the test below?

Reply:
no, I missed adding a `// TO DO` comment.